src/arrayMethodSort.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    if (this.length < 2) {
+      return this;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < (this.length - i - 1); j++) {
+        if ((compareFunction !== undefined
+          && (compareFunction(this[j], this[j + 1]) > 0))
+          || (compareFunction === undefined
+          && (this[j].toString() > this[j + 1].toString()))) {
```

Comment:
Consider thinking about how to improve the readability of these conditionals and move them to separate variables.

Reply:
Done